src/ft2font_wrapper.cpp

```diff
@@ -381,115 +376,84 @@ static unsigned long read_from_file_callback(FT_Stream stream,
+        if (!count) {
+            return 1;  // Non-zero signals error, when count == 0.
+        }
+    }
+    return n_read;
 }
 
 static void close_file_callback(FT_Stream stream)
 {
-    PyFT2Font *def = (PyFT2Font *)stream->descriptor.pointer;
-
-    if (mpl_PyFile_DupClose(def->py_file, def->fp, def->offset)) {
-        throw std::runtime_error("Couldn't close file");
+    PyObject *py_file = ((PyFT2Font *)stream->descriptor.pointer)->py_file;
+    PyObject *close_result = NULL;
+    if (!(close_result = PyObject_CallMethod(py_file, "close", ""))) {
+        goto exit;
     }
-
-    if (def->close_file) {
-        mpl_PyFile_CloseFile(def->py_file);
+exit:
+    Py_XDECREF(close_result);
+    Py_DECREF(py_file);
+    if (PyErr_Occurred()) {
+        PyErr_WriteUnraisable(py_file);
     }
-
-    Py_DECREF(def->py_file);
-    def->py_file = NULL;
 }
 
 static int convert_open_args(PyFT2Font *self, PyObject *py_file_arg, FT_Open_Args *open_args)
 {
-    PyObject *py_file = NULL;
-    int close_file = 0;
-    FILE *fp;
+    PyObject *open = NULL;
     PyObject *data = NULL;
-    char *data_ptr;
-    Py_ssize_t data_len;
-    long file_size;
-    FT_Byte *new_memory;
-    mpl_off_t offset = 0;
 
     int result = 0;
 
     memset((void *)open_args, 0, sizeof(FT_Open_Args));
 
     if (PyBytes_Check(py_file_arg) || PyUnicode_Check(py_file_arg)) {
-        if ((py_file = mpl_PyFile_OpenFile(py_file_arg, (char *)"rb")) == NULL) {
+        if (!(open = PyDict_GetItemString(PyEval_GetBuiltins(), "open"))  // Borrowed reference.
+            || !(self->py_file = PyObject_CallFunction(open, "Os", py_file_arg, "rb"))) {
             goto exit;
         }
-        close_file = 1;
+        self->close_file = 1;
+    } else if (!PyObject_HasAttrString(py_file_arg, "read")
+               || !(data = PyObject_CallMethod(py_file_arg, (char *)"read", (char *)""))
+               || !PyBytes_Check(data)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "First argument must be a path or binary-mode file object");
+        goto exit;
     } else {
+        self->py_file = py_file_arg;
         Py_INCREF(py_file_arg);
-        py_file = py_file_arg;
-    }
-
-    if ((fp = mpl_PyFile_Dup(py_file, (char *)"rb", &offset))) {
-        Py_INCREF(py_file);
-        self->py_file = py_file;
-        self->close_file = close_file;
-        self->fp = fp;
-        self->offset = offset;
-        fseek(fp, 0, SEEK_END);
-        file_size = ftell(fp);
-        fseek(fp, 0, SEEK_SET);
-
-        self->stream.base = NULL;
-        self->stream.size = (unsigned long)file_size;
-        self->stream.pos = 0;
-        self->stream.descriptor.pointer = self;
-        self->stream.read = &read_from_file_callback;
-        self->stream.close = &close_file_callback;
-
-        open_args->flags = FT_OPEN_STREAM;
-        open_args->stream = &self->stream;
-    } else {
-        if (PyObject_HasAttrString(py_file_arg, "read") &&
```

Comment:
... or alternatively we can just read the whole file in memory, as previously the case in this branch...